pkg/gatewayserver/scheduling/scheduler.go

```diff
@@ -83,41 +88,75 @@ func NewScheduler(ctx context.Context, fp *frequencyplans.FrequencyPlan, enforce
 		scheduleAnytimeDelay = &ScheduleTimeShort
 	}
 
-	toa := fp.TimeOffAir
+	var toa *frequencyplans.TimeOffAir
+	for _, fp := range fps {
+		if toa != nil && fp.TimeOffAir != *toa {
+			return nil, errFrequencyPlansTimeOffAir
+		}
+		toa = &fp.TimeOffAir
+	}
+
 	if toa.Duration < QueueDelay {
 		toa.Duration = QueueDelay
 	}
+
 	s := &Scheduler{
 		clock:                &RolloverClock{},
-		respectsDwellTime:    fp.RespectsDwellTime,
-		timeOffAir:           toa,
+		timeOffAir:           *toa,
+		fps:                  fps,
 		timeSource:           timeSource,
 		scheduleAnytimeDelay: *scheduleAnytimeDelay,
 	}
 	if enforceDutyCycle {
-		if subBands := fp.SubBands; len(subBands) > 0 {
-			for _, subBand := range subBands {
-				params := SubBandParameters{
-					MinFrequency: subBand.MinFrequency,
-					MaxFrequency: subBand.MaxFrequency,
-					DutyCycle:    subBand.DutyCycle,
+		for _, fp := range fps {
+			if subBands := fp.SubBands; len(subBands) > 0 {
+				for _, subBand := range subBands {
+					params := SubBandParameters{
+						MinFrequency: subBand.MinFrequency,
+						MaxFrequency: subBand.MaxFrequency,
+						DutyCycle:    subBand.DutyCycle,
+					}
+					sb := NewSubBand(ctx, params, s.clock, nil)
+					var isIdentical bool
+					for _, subBand := range s.subBands {
+						if subBand.IsIdentical(sb) {
+							isIdentical = true
+							break
```

Comment:
`continue` and skip check on L126

Reply:
Hmm why? I also want to check if there's an overlap which is on `L126`